debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeTest.java

```diff
@@ -199,6 +201,43 @@ public void shouldGenerateRecordForUpdateEvent() throws InterruptedException {
         assertThat(value.schema().fields()).hasSize(2);
     }
 
+
+    @Test
+    public void shouldGenerateRecordForDeleteEventWithoutNullRecord() throws InterruptedException {
```

Comment:
The name seems misleading, because the tombstone *is* created but then it's dropped by the SMT, right?

Reply:
No, the tombstone is not created at all here because the producer is being run with `Configuration configurationTemp = Configuration.create().with(MongoDbConnectorConfig.COMAPCT_DELETE_OPERATIONS, "false").build();`
And it is being checked in this assertion `assertThat(produced.size()).isEqualTo(1)`.

The second test `shouldGenerateRecordForDeleteEvent()`, which I haven touched, tests the existing behaviour, where `assertThat(produced.size()).isEqualTo(2);` is asserted.

Reply:
@gunnarmorling 
Just to add to my previous answer, I've only added `shouldGenerateRecordForDeleteEventWithoutNullRecord()` test because the existing one `shouldGenerateRecordForDeleteEvent()` had `assertThat(produced.size()).isEqualTo(2);` assertion and I had to change it to `isEqualTo(1)` in my test (because only one record is being generated).

I had a deeper dive and here are my findings (please correct me if I'm wrong): 
`shouldGenerateRecordForDeleteEvent()` emits 2 messages, SMT converts the first into tombstone and SHOULD(?) drop the second one, for consistency, as it is being done in `UnwrapFromEnvelope`, but unfortunately, as far as I can tell `UnwrapFromMongoDbEnvelope` doesn't drop the second record. Also please notice that the bit to test this is missing; so although `produced.size()` is equal to, only the the first record is under test `SourceRecord record = produced.get(0);` e.g.:
If I add the below to `shouldGenerateRecordForDeleteEvent()`:
```
SourceRecord record2 = produced.get(1);
SourceRecord transformed2 = transformation.apply(record);
assertThat(transformed2).isNull();
```
will produce:
`java.lang.AssertionError: <SourceRecord{sourcePartition={rs=rs0, server_id=serverX.}, sourceOffset={sec=1000, ord=1, h=12345678}} ConnectRecord{topic='serverX...dbA.c1', kafkaPartition=null, key=Struct{id=5a6f92219ba5bc5ca04bcf15}, value=null, timestamp=null}> should be null`  


Moreover these tests are slightly broader than normal UT because they also partially test the producers by checking the `produced.size()` which IMHO is no required here - the `RecordMarkersTest` does that already, please correct me if I'm wrong.